packages/office-ui-fabric-react/src/components/Link/Link.base.tsx

```diff
@@ -74,4 +65,50 @@ export class LinkBase extends BaseComponent<ILinkProps, any> implements ILink {
       onClick(ev);
     }
   }
+
+  private _removeInvalidPropsForRootType(RootType: string | React.ComponentClass | React.StatelessComponent, props: ILinkProps): Partial<ILinkProps> {
+    // Deconstruct the props so we remove props like `renderAs`, `theme` and `getStyles`
+    // as those will always be removed. We also take some props that are optional
+    // based on the RootType.
+    const {
```

Comment:
I get what's happening here, but I'm nervous about doing this every time we use `renderAs`. I definitely believe removing some things (like `renderAs`) should be stripped to kill potentially circular rendering scenarios...

Reply:
I don' have a counter proposal other than only removing the absolute necessary things